src/components/common/Layout/components/Account/components/Notifications/components/xmtp.ts

```diff
@@ -0,0 +1,142 @@
+import { useToast } from "@chakra-ui/react"
+import { Client, useCanMessage } from "@xmtp/react-sdk"
+import useUser from "components/[guild]/hooks/useUser"
+import useShowErrorToast from "hooks/useShowErrorToast"
+import { useCallback, useEffect, useState } from "react"
+import { useFetcherWithSign } from "utils/fetcher"
+import { useAccount, useWalletClient } from "wagmi"
+
+export const useSaveXmtpKeys = () => {
+  const fetcherWithSign = useFetcherWithSign()
+  const { data: signer } = useWalletClient()
+
+  const { id } = useUser()
+
+  return useCallback(
```

Comment:
I think we could use the custom `useSubmit` hook here, it would handle the loading/error state automatically ;)

Reply:
done, i put the saving functions into the submit hook